### PR TITLE
Cleanup: Move IntervalType from merge-tree to sequence

### PR DESCRIPTION
**Pull request by anthony-murphy**
Doing some cleanup and came across and unused type in the merge-tree package. moving it to the sequence package where it is used.

### CODE REVIEW

File: packages/dds/sequence/src/intervalCollection.ts [hunk 1]
```diff
@@ -35,6 +34,14 @@ import { IValueFactory, IValueOpEmitter, IValueOperation, IValueType } from "./m
 
 const reservedIntervalIdKey = "intervalId";
 
+
+export enum IntervalType {
```
Comment (ChumpChief) [inline packages/dds/sequence/src/intervalCollection.ts:38]:
```suggestion
const reservedIntervalIdKey = "intervalId";

export enum IntervalType {
```

File: packages/dds/sequence/src/intervalCollection.ts [hunk 2]
```diff
@@ -414,7 +421,7 @@ export class LocalIntervalCollection<TInterval extends ISerializableInterval> {
         private readonly label: string,
         private readonly helpers: IIntervalHelpers<TInterval>) {
         this.endIntervalTree =
-            // eslint-disable-next-line @typescript-eslint/unbound-method
+
             new RedBlackTree<TInterval, TInterval>(helpers.compareEnds);
```
Comment (ChumpChief) [inline packages/dds/sequence/src/intervalCollection.ts:425]:
```suggestion
            new RedBlackTree<TInterval, TInterval>(helpers.compareEnds);
```

File: packages/dds/merge-tree/src/test/testClient.ts [hunk 3]
```diff
@@ -259,7 +259,7 @@ export class TestClient extends Client {
         while (start < this.getLength()) {
             chunk = this.getText(start, start + TestClient.searchChunkSize);
 
-            // eslint-disable-next-line @typescript-eslint/prefer-regexp-exec
+
             const result = chunk.match(target);
```
Comment (ChumpChief) [inline packages/dds/merge-tree/src/test/testClient.ts:263]:
```suggestion
            const result = chunk.match(target);
```

File: packages/dds/merge-tree/src/test/testClient.ts [hunk 4]
```diff
@@ -78,7 +78,7 @@ export class TestClient extends Client {
         specToSeg: (spec: IJSONSegment) => ISegment): Promise<TestClient> {
         const client2 = new TestClient(undefined, specToSeg);
         const { catchupOpsP } = await client2.load(
-            // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+
             {
```
Comment (ChumpChief) [inline packages/dds/merge-tree/src/test/testClient.ts:82]:
```suggestion
            {
```

**Review by ChumpChief — APPROVED** [review]
Seems good to me - I think this also implies that `IntervalType` isn't really a generic set of interval types to be reused in other merge tree consumers (as its prior home would imply), but rather is a set of interval types specific to shared-string and not really intended for reuse?

File: packages/dds/sequence/src/intervalCollection.ts [hunk 5]
```diff
@@ -414,7 +421,7 @@ export class LocalIntervalCollection<TInterval extends ISerializableInterval> {
         private readonly label: string,
         private readonly helpers: IIntervalHelpers<TInterval>) {
         this.endIntervalTree =
-            // eslint-disable-next-line @typescript-eslint/unbound-method
+
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/intervalCollection.ts:424]:
```suggestion
```

File: BREAKING.md [hunk 6]
```diff
@@ -10,6 +10,16 @@ There are a few steps you can take to write a good change note and avoid needing
 - Provide guidance on how the change should be consumed if applicable, such as by specifying replacement APIs.
 - Consider providing code examples as part of guidance for non-trivial changes.
 
+## 0.58 Breaking changes
+- [Move IntervalType from merge-tree to sequence package](#Move-IntervalType-from merge-tree-to-sequence-package)
+
+### Move IntervalType from merge-tree to sequence package
+Move the type from the merge-tree package where it isn't used to the sequence where it is used
```
Comment (anthony-murphy) [inline BREAKING.md:17]:
```suggestion
Move the type from the merge-tree package where it isn't used to the sequence package where it is used
```

**Pull request merged**